### PR TITLE
Fix some test incompatibilities with latest numpy

### PR DESCRIPTION
**Pull request by Strilanc**
- Also fix ZPowGate and PhasedXPowGate not supporting approximate equality
- Also fix some tests using deprecated methods

### CODE REVIEW

File: cirq/ops/common_gates.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def __repr__(self) -> str:
         ).format(proper_repr(self._exponent), self._global_shift)
 
 
-@value.value_equality
```
Comment (dabacon) [inline cirq/ops/common_gates.py:380]: so the issue here is that this override eigen_gate's decorator?  Do we need to do this for XPowGate and YPowGate?

Comment (Strilanc) [reply cirq/ops/common_gates.py]: Actually.. turns out the test I added passed anyways. Removed this change.

Comment (dabacon) [reply cirq/ops/common_gates.py]: OK, I think there is still a bug here from one of the cases I found, but will follow up separately.

**Comment by dabacon** [comment]
api test doesn't like you :)

**Pull request merged**